py/dml/types.py

```diff
@@ -61,6 +66,16 @@ def __init__(self, t):
     def __str__(self):
         return "Unknown type: %s" % (self.type,)
 
+class DMLUnkeyableType(DMLTypeError):
+    def __init__(self, t, clarification=None):
+        DMLTypeError.__init__(self)
+        self.type = t
+        self.clarification = clarification
+    def __str__(self):
+        return "Unkeyable type: %s%s" % (self.type,
+                                         ('; ' + self.clarification)
```

Comment:
1. punctuation: I think `;` is inaccurate here, perhaps `Unkeyable type '{type}': {clarification}`? (And no need to make clarification optional)

Reply:
ping